coap-gateway/service/auth.go

```diff
@@ -68,3 +27,47 @@ func NewAuthInterceptor() Interceptor {
 		return ctx, expire.IsValid()
 	}
 }
+
+type jwtClaims jwt.MapClaims
+
+func (s *Service) ValidateToken(ctx context.Context, token string) (jwtClaims, error) {
+	ctx, cancel := context.WithTimeout(ctx, s.config.APIs.COAP.KeepAlive.Timeout)
+	defer cancel()
+	m, err := s.jwtValidator.ParseWithContext(ctx, token)
+	if err != nil {
+		return nil, err
+	}
+	return jwtClaims(m), nil
+}
+
+/// Get expiration time (exp) from user info map.
+/// It might not be set, in that case zero time and no error are returned.
+func (u jwtClaims) getExpirationTime() (time.Time, error) {
+	const expKey = "exp"
+	v, ok := u[expKey]
+	if !ok {
+		return time.Time{}, nil
+	}
+
+	exp, ok := v.(float64) // all integers are float64 in json
+	if !ok {
+		return time.Time{}, fmt.Errorf("invalid jwtClaims: invalid %v value type", expKey)
+	}
+	return pkgTime.Unix(int64(exp), 0), nil
+}
+
+/// Validate that ownerClaim is set and that it matches given user ID
+func (u jwtClaims) validateOwnerClaim(ocKey string, userID string) error {
+	v, ok := u[ocKey]
+	if !ok {
+		return fmt.Errorf("invalid jwtClaims: %v not set", ocKey)
```

Comment:
jwtClaims are not invalid. They are okay. Be more concrete in log messages please.
"configured owner claim %v is not present in the device token" where %v is value from configuration -> ownerClaim